vyper/semantics/validation/module.py

```diff
@@ -210,8 +211,12 @@ def visit_AnnAssign(self, node):
         if is_constant:
             if not node.value:
                 raise VariableDeclarationException("Constant must be declared with a value", node)
-            if not check_literal(node.value):
+            if not check_constant(node.value):
                 raise StateAccessViolation("Value must be a literal", node.value)
+            if isinstance(node.value, vy_ast.Attribute):
+                # Check for environment variables
+                if node.value.get("value.id") in get_constant_vars():
```

Comment:
maybe we can remove this now that we separated the behavior of `check_constant` and `check_kwargable`